core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -770,9 +794,19 @@ private[spark] class TaskSetManager(
         logError("Unknown TaskEndReason: " + e)
         None
     }
-    // always add to failed executors
-    failedExecutors.getOrElseUpdate(index, new HashMap[String, Long]()).
-      put(info.executorId, clock.getTimeMillis())
+
+    // we might rack up a bunch of fetch-failures in rapid succession, due to a bad node.  But
+    // that bad node will get handled separately by spark's stage-failure handling mechanism.  It
+    // shouldn't penalize *this* executor at all, so don't count it as a task-failure as far as
+    // the blacklist is concerned.
+    val countTowardsTaskFailures = reason match {
+      case fail: TaskFailedReason => fail.countTowardsTaskFailures
```

Comment:
yeah, agree.  There is also the code on the old line 788 which made it seem like it was possible for it to be something else, which is what this line is replacing.  I wondered about doing this, but decided for the smaller change, though I'm fine either way.  This means we effectively move the cast to [`TaskResultGetter.enqueueFailedTask`](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/scheduler/TaskResultGetter.scala#L128).  I will make that change.